src/transport.rs

```diff
@@ -285,25 +220,27 @@ impl EccTransport {
         }
         bit_field
     }
-    
-    fn decode_swi_to_uart(&mut self, swi_msg: &BytesMut, uart_msg: &mut BytesMut ) {
-    
-        uart_msg.clear();
-        assert!( (swi_msg.len() % 8) == 0);
-        uart_msg.resize( &swi_msg.len() / 8, 0 );
-    
-        let mut i = 0; 
-        for byte in uart_msg.iter_mut() {
-            let bit_slice= &swi_msg[i..i+8];
-            
-            for bit in bit_slice.iter(){
-                if *bit == 0x7F || *bit == 0x7E {
-                    *byte ^= 1;
+
+    fn decode_swi_to_uart(&mut self, buf: &mut [u8]) -> Result {
+        for byte_idx in 0..buf.len() {
+            let mut decoded_byte = 0;
+            let mut bit_mask: u8 = 1;
+
+            while bit_mask != 0 {
+                let mut rx_byte = [0; 1];
```

Comment:
Should it be [0,1] for array declaration ?

Reply:
No this is how rust does array initialization. It means an array of length 1 with all items initialized to 0

Reply:
Yes you are correct. Just checked it. Thanks for clarification.
https://doc.rust-lang.org/rust-by-example/primitives/array.html